File: src/types/changelog.ts
export type ChangeType = 'prompt_update' | 'model_change' | 'tool_update' | 'config_change';

export interface Changelog {
  id: string;
  product_id: string | null;
  experiment_id: string | null;
  change_type: ChangeType;
  title: string;
  description: string | null;
  before_snapshot_id: string | null;
  after_snapshot_id: string | null;
  impact_assessment: string | null;
  created_at: string;
  updated_at: string;
}

export interface ChangelogCreate {
  product_id?: string;
  experiment_id?: string;
  change_type: ChangeType;
  title: string;
  description?: string;
  before_snapshot_id?: string;
  after_snapshot_id?: string;
  impact_assessment?: string;
}

File: app/schemas/changelog.py
from datetime import datetime
from typing import Optional
from uuid import UUID
from pydantic import BaseModel, ConfigDict


class ChangelogCreate(BaseModel):
    product_id: Optional[UUID] = None
    experiment_id: Optional[UUID] = None
    change_type: str
    title: str
    description: Optional[str] = None
    before_snapshot_id: Optional[UUID] = None
    after_snapshot_id: Optional[UUID] = None
    impact_assessment: Optional[str] = None


class ChangelogResponse(BaseModel):
    id: UUID
    product_id: Optional[UUID] = None
    experiment_id: Optional[UUID] = None
    change_type: str
    title: str
    description: Optional[str] = None
    before_snapshot_id: Optional[UUID] = None
    after_snapshot_id: Optional[UUID] = None
    impact_assessment: Optional[str] = None
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)
